packages/app/src/cli/services/app_logs/fetch-app-logs.ts

```diff
@@ -0,0 +1,89 @@
+import {writeAppLogsToFile} from './write-app-logs.js'
+import {partnersFqdn} from '@shopify/cli-kit/node/context/fqdn'
+import {Writable} from 'stream'
+
+const pollingInterval = 450
+
+const generateFetchAppLogUrl = async (cursor?: string) => {
+  const fqdn = await partnersFqdn()
+  const url = `https://${fqdn}/app_logs/poll`
+  return url + (cursor ? `?cursor=${cursor}` : '')
+}
+
+export interface AppEventData {
+  shop_id: number
+  api_client_id: number
+  payload: string
+  event_type: string
+  cursor: string
+  status: 'success' | 'failure'
+  log_timestamp: string
+}
+
+export const fetchAppLogs = async ({
+  stdout,
+  appLogsFetchInput: {jwtToken, cursor},
+  apiKey,
+}: {
+  stdout: Writable
+  appLogsFetchInput: {jwtToken: string; cursor?: string}
+  apiKey: string
+}) => {
+  const url = await generateFetchAppLogUrl(cursor)
+  const response = await fetch(url, {
+    method: 'GET',
+    headers: {
+      Authorization: `Bearer ${jwtToken}`,
+    },
+  })
+
+  if (!response.ok) {
+    // We should add some exponential backoff here to not spam partners
+    // additional error details would be good as well
+
+    // if the error is about the token being expired,
+    // we should perform the refresh here instead of requiring restarting app dev
+
+    throw new Error(`HTTP error! status: ${response.status}`)
+  }
+
+  const data = (await response.json()) as {
+    app_logs?: AppEventData[]
+    cursor?: string
+    errors?: string[]
+  }
+
+  if (data?.errors) {
+    data?.errors?.forEach((error) => {
+      stdout.write(`${error}\n`)
+    })
+  } else if (data.app_logs) {
+    const {app_logs: appLogs} = data
+
+    await Promise.all(
+      appLogs.map(async (log) => {
+        await writeAppLogsToFile({
+          appLog: log,
+          apiKey,
+          stdout,
+        })
+      }),
+    )
+
+    // print the logs from the appLogs as well
+  }
+
+  const cursorFromResponse = data?.cursor
+
+  // eslint-disable-next-line @typescript-eslint/no-misused-promises
```

Comment:
🤔 Swapped it around to respect the lint, but I feel it's a bit weird?
First time around, we invoke fetchAppLogs directly, so if it errors out, we'd get the message from it directly.
On the second time around, would an error "bubble up" to this setTimeout's catch, and then be handled there?

Feels like the way I've set it up is wrong since ideally the error flow would always go the same way?